### PR TITLE
feat: Eliminating use of @dynamicReplacement, auto generate boilerplate, update required dependencies

### LINKED ISSUE #110: Simplify boilerplate and get rid of dynamicReplacement if possible
As [suggested](https://github.com/apple/swift-nio/pull/2392#issuecomment-1471965913), maybe this can help us kill @dynamicReplacement:
 
> Have you tried exploring calling the same methods as Argument Parser does but just from your library, e.g. you could provide a protocol called Benchmark that has a main method that looks like this
```
  public static func main() async {
    // Setup everything that benchmark needs
    do {
      var command = try parseAsRoot()
      if var asyncCommand = command as? AsyncParsableCommand {
        try await asyncCommand.run()
      } else {
        try command.run()
      }
    } catch {
      exit(withError: error)
    }

  // Record benchmark results etc.
  }
```
> Since all methods are public in Argument Parser you should be able to replicate anything the package is doing in your own main method. Maybe I am missing something here though!

### PR DESCRIPTION
**Pull request by hassila**
## Description

Cleaning up API for stable release:

* Add boilerplate generation plugin to remove boilerplate needed for benchmark suite definitions
* Change benchmark boilerplate to a simple 'let benchmarks = { }'
* Change target name from BenchmarkSupport -> Benchmark for external users
* Need to change dependency to have both Benchmark and BenchmarkPlugin for external users

Fixes #110 

## How Has This Been Tested?

Manual and test suite

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [x] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [x] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Plugins/BenchmarkBoilerplateGenerator/BenchmarkBoilerplateGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
+//
+// Copyright (c) 2023 Ordo One AB.
```
Comment (dimlio) [inline Plugins/BenchmarkBoilerplateGenerator/BenchmarkBoilerplateGenerator.swift:2]:
```suggestion
// Copyright (c) 2023 Ordo One AB
```
:)

File: Plugins/BenchmarkBoilerplateGenerator/BenchmarkBoilerplateGenerator.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+struct Benchmark: AsyncParsableCommand {
+    @Option(name: .long, help: "Name of the target")
+    var target: String
+
+    @Option(name: .long, help: "Output file path")
+    var output: String
+
+    mutating func run() async throws {
+        let outputPath = FilePath(output) // package
+        var boilerplate = """
```
Comment (dimlio) [inline Plugins/BenchmarkBoilerplateGenerator/BenchmarkBoilerplateGenerator.swift:24]: Can't be `let`?

File: README.md [hunk 3]
```diff
@@ -41,19 +41,17 @@ There are [documentation available](https://swiftpackageindex.com/ordo-one/packa
...
+import Benchmark
 
+let benchmarks = {
     Benchmark("Minimal benchmark") { benchmark in
       // measure something here
     }
 
     Benchmark("All metrics, full concurrency, async",
+              configuration: .init(
               metrics: BenchmarkMetric.all,
```
Comment (dimlio) [inline README.md:53]: Maybe indent here?

**Review by dimlio — APPROVED** [review]

File: Plugins/BenchmarkBoilerplateGenerator/BenchmarkBoilerplateGenerator.swift (hunk 2 above)
Comment (hassila) [reply Plugins/BenchmarkBoilerplateGenerator/BenchmarkBoilerplateGenerator.swift:24]: Nope, `boilerplate.withUTF8` requires a mutable var.

**Pull request merged**